src/lib/concurrency/transaction_context.cpp

```diff
@@ -189,16 +176,13 @@ void TransactionContext::_wait_for_active_operators_to_finish() const {
   _active_operators_cv.wait(lock, [&] { return _num_active_operators != 0; });
 }
 
-bool TransactionContext::_transition(TransactionPhase from_phase, TransactionPhase to_phase,
+void TransactionContext::_transition(TransactionPhase from_phase, TransactionPhase to_phase,
                                      TransactionPhase end_phase) {
   auto expected = from_phase;
   const auto success = _phase.compare_exchange_strong(expected, to_phase);
 
-  if (success) {
-    return true;
-  } else {
+  if (!success) {
     Assert((expected == to_phase) || (expected == end_phase), "Invalid phase transition detected.");
```

Comment:
I don't understand these cases. Why is it ok if the phase is in those states? Also, what is the "end phase" anyway? Can you doc that?

Reply:
I don't think this is part of this PR, that code was there before. Essentially what this is testing is multiple people calling commit() - something we're not sure we want, anyway. Only one of them should perform the transition and everyone else checks whether the the transaction is one of the expected follow-up states.

Reply:
> I don't think this is part of this PR

That is rarely a good reason. Either fix it in this PR or open an issue.

> Essentially what this is testing is multiple people calling commit()

I'd make this illegal and see if it causes any problems.